### ISSUE TITLE
1.6.2 network connectivity "coma" issue

### ISSUE THREAD

**Issue by Aterfax** [zerotier/ZeroTierOne#1334]
## Problem description

Routes do not appear to be functioning correctly on the Windows 10 Home device making it unable to talk to other devices. (Tested with firewall off.)

- What you expect to be happening.

Devices can consistently ping each other and connect to each other.

- Any steps to reproduce the error.

Unclear? Use Windows 10 Home? that appears not to respect routes?

## Additional information 

Network is setup as follows :  
 - Range - 192.168.193.* 
 - Route - 192.168.193.0/24

3 indicated devices share the same local network (10.0.10.X) and can ping / connect to one another.

**192.168.193.247:**
 - OS: [Ubuntu 20.04.1 LTS]
 - ZeroTier Version 1.6.2
 - Local network and pingable over it.
 
**192.168.193.79:**
 - OS: [Windows 10 Pro 20H2]
 - ZeroTier Version 1.6.2
 - Local network and pingable over it.
 
 **192.168.193.111:**  (Device seeming to have issues.)
 - OS: [Windows 10 Home 20H2]
 - ZeroTier Version 1.6.2
 - Local network and pingable over it.
 
  **192.168.193.70:**
 - OS: [Centos 7.5]
 - ZeroTier Version 1.6.2
 - Remote network

Pingable from 192.168.193.79
192.168.193.247  - Yes
192.168.193.111 - Yes
192.168.193.70 - Yes

Pingable from 192.168.193.247
192.168.193.79  - Yes
**192.168.193.111 - No**
192.168.193.70 - Yes

Pingable from 192.168.193.111
192.168.193.79 - Yes
**192.168.193.247 - No**
**192.168.193.70 - No**

Pingable from 192.168.193.70
192.168.193.79 - Yes
192.168.193.247 -Yes
**192.168.193.111 - No**

**Additional context**
- Router Config: are you permitting port 9993, uPnP, and NAT-PMP? Yes
- Firewall Config: Firewall off where appropriate.

Routes appear to be partially broken from device 192.168.193.111 as tracerouting as below shows the route going via the default gateway for .247 (and .70 unshown) but working correctly for .79

Not sure if I am having some peer connectivity issues from 192.168.193.111 (this would be odd given .111 is in the same network as .79)

```
tracert 192.168.193.247

Tracing route to 192.168.193.247 over a maximum of 30 hops

  1    <1 ms    <1 ms    <1 ms  gateway.domain.com [10.0.10.254]
  2     9 ms     9 ms     9 ms  host-62-24-254-204.as13285.net [62.24.254.204]
  3    10 ms     9 ms     9 ms  host-78-144-1-63.as13285.net [78.144.1.63]
  4     *        *        *     Request timed out.
  5     *        *        *     Request timed out.
  6     *        *        *     Request timed out.
  7     *        *        *     Request timed out.

```

```

tracert 192.168.193.79


Tracing route to Device [192.168.193.79]
over a maximum of 30 hops:

  1     1 ms     1 ms     1 ms device [192.168.193.79]



```

192.168.193.111 routes are as follows:
```
IPv4 Route Table
===========================================================================
Active Routes:
Network Destination        Netmask          Gateway       Interface  Metric
          0.0.0.0          0.0.0.0      10.0.10.254      10.0.10.X    110
          0.0.0.0          0.0.0.0   25.255.255.254  192.168.193.111  10034
        10.0.10.0    255.255.255.0         On-link       10.0.10.X 331
      10.0.10.X  255.255.255.255         On-link       10.0.10.X    331
      10.0.10.255  255.255.255.255         On-link       10.0.10.X    331
        127.0.0.0        255.0.0.0         On-link         127.0.0.1    331
        127.0.0.1  255.255.255.255         On-link         127.0.0.1    331
  127.255.255.255  255.255.255.255         On-link         127.0.0.1    331
    192.168.193.0    255.255.255.0         On-link   192.168.193.111    291
  192.168.193.111  255.255.255.255         On-link   192.168.193.111    291
  192.168.193.255  255.255.255.255         On-link   192.168.193.111    291
        224.0.0.0        240.0.0.0         On-link         127.0.0.1    331
        224.0.0.0        240.0.0.0         On-link       10.0.10.X    331
        224.0.0.0        240.0.0.0         On-link   192.168.193.111    291
  255.255.255.255  255.255.255.255         On-link         127.0.0.1    331
  255.255.255.255  255.255.255.255         On-link       10.0.10.X    331
  255.255.255.255  255.255.255.255         On-link   192.168.193.111    291

===========================================================================
```

Interface metrics as follows for 192.168.193.111
```
ifIndex InterfaceAlias                  AddressFamily NlMtu(Bytes) InterfaceMetric Dhcp     ConnectionState PolicyStore
------- --------------                  ------------- ------------ --------------- ----     --------------- -----------
31      ZeroTier One [XXXXXXXX]          IPv6                  2800              35 Enabled  Connected       ActiveStore
14      Ethernet                         IPv6                  1488              75 Enabled  Connected       ActiveStore
1       Loopback Pseudo-Interface 1     IPv6            4294967295              75 Disabled Connected       ActiveStore
31      ZeroTier One [XXXXXXXX]          IPv4                  2800              35 Disabled Connected       ActiveStore
14      Ethernet                           IPv4                  1500              75 Enabled  Connected       ActiveStore
1       Loopback Pseudo-Interface 1     IPv4            4294967295              75 Disabled Connected       ActiveStore

```
I am perplexed as this should be working correctly,everything is working correctly for the other Windows 10 Pro device (192.168.193.79) and there's some inconsistency in somehow 192.168.193.111 being able to ping 192.168.193.79.

I've confirmed the UPNP ports are requested and open for each device on the router also.

**Comment by EugeneKorneev** [comment]
I have similar issue
I cant ssh to new device in network from my notebook
But i can ssh via other

**Comment by hardwareadictos** [comment]
Similar issue reported here: https://github.com/zerotier/ZeroTierOne/issues/1343

No answer from ZeroTier..............................

**Comment by sbilly** [comment]
node1, node2, node3 are 3 nodes on same LAN. They are not the same network, but can ping each other via physical IP address while node2 can not reach node1 via overlay IP.

```
node1:
OS: Ubuntu 16.04 LTS
Physical IP: 192.168.31.150/24
Overlay IP: 10.147.18.100

node2:
OS: Ubuntu CentOS 7
Physical IP: 192.168.31.160/24
Overlay IP: 10.147.18.94

node3:
OS: macOS X latest
Physical IP: 192.168.0.101/24
Overlay IP: 10.147.18.1
```

ping results:

```
Physical IP:
node1 <-> node2, works
node1 <-> node3, works
node2 <-> node3, works

Overlay IP:
node1 <-> node2, not working
node1 <-> node3, works
node2 <-> node3, works
```

I try to using tcpdump listen on the node1's default nic(the only one nic connect to interenet) while ping node2's overlay IP on node1. tcpdump can't capture any packets.

**Comment by lastb0isct** [comment]
I've been having this issue for weeks now. OSX system works with zerotier hosts outside the network. But windows system on the same network can't hit any of them, even the OSX on the same LAN.

We need someone to take a look at this. This is functionally broken.

**Comment by isakrubin** [comment]
@lastb0isct Had similar issues, but was solved by this https://superuser.com/questions/1137912/ping-to-windows-10-not-working-if-file-and-printer-sharing-is-turned-off

**Comment by jkwmoore** [comment]
Yeah this isn't just ping - there is no peer connectivity - I am trying to SSH between devices.

**Comment by Aterfax** [comment]
Yeah this isn't just ping - there is no peer connectivity - I am trying to SSH between devices.

**Comment by lastb0isct** [comment]
> @lastb0isct Had similar issues, but was solved by this https://superuser.com/questions/1137912/ping-to-windows-10-not-working-if-file-and-printer-sharing-is-turned-off
> 
> Not sure it's related to your case, but worth a try

I have no firewall turned on. No connectivity at all as @Aterfax said, it's not only ping

**Comment by aadnehovda** [comment]
I have similar issues, in this case there is total loss of connectivity between only two of the nodes in the network (which consists of around five Linux dmvpn gateways as part of the intranet OSPF domain). I can reach either of the "problematic" nodes (ping, ssh) from all the other nodes. A restart of the zerotier service fixes the issue, but after a while traffic stops flowing again. At 1.6.2 but upgrading to 1.6.3 now to see if it improves.

**Comment by sbilly** [comment]
It seems fixed by 1.6.4.

**Comment by Aterfax** [comment]
> 
> 
> It seems fixed by 1.6.4.

I have re-installed also and this appears to be the case - although it may have been simply the act of purging / reinstalling.

**Comment by joseph-henry** [comment]
Hey everyone. Just checking in, are you still seeing an improvement with 1.6.4?

**Comment by joseph-henry** [comment]
I've created a new experimental branch with some reverted path logic:

[experimental-path-fix](https://github.com/zerotier/ZeroTierOne/tree/experimental-path-fix)

If anyone is still having issues, delete your `peers.d` directory and try using this new branch. It would be helpful if you could build with `ZT_DEBUG=1` and try each of the latest two commits one by one and give feedback on what works.

We appreciate your help and patience.

**Comment by garret** [comment]
Can you confirm the issue I am having is the same described here? If yes, how could I help? Would it be possible to have some executables ready for installation for Windows? 

> I am using ZeroTier on several devices (Android, Linux and Windows). No problems with Android and Linux but on two Windows 10 laptops (using version 1.6.4) sitting on two different locations sometimes I cannot connect to my network after a reboot.
> 
> Despite on the ZeroTier config webpage I can see the devices as online, the Windows application cannot show I am connected to the address of my network and that is also confirmed by not being able to ping devices on my ZeroTier network.
> 
> Restarting the app is not enough to fix. I have to first uninstall and then reinstall again. I have disabled Win 10 firewall and don't have any extra antivirus/firewall applications installed.

**Comment by Thingamajig** [comment]
Hey Team,

I still seem to be running into the coma issue over on this side on primarily Windows-based nodes even with the latest 1.6.4 patch on both sides. 

Is there anything you'd need from me in the way of logs etc? I can't consistently replicate the issue which makes it difficult to diagnose further. 

What we do know is that it happens regardless of Windows OS version, Win 7 through to Win 10. As per the commenter above, any chance we can have some Windows executables to test with?

Thanks :)

**Comment by joseph-henry** [comment]
@Thingamajig Did you try current `dev` branch? That contains a fix that is likely to fix this.

**Comment by garret** [comment]
Where can I find the `dev` branch windows executables?

**Comment by joseph-henry** [comment]
Closing this ticket soon since we believe this issue has been resolved in `1.6.5`. Please let us know here if that is not the case for you.
